### ISSUE TITLE
comma-spacing conflicts with array-bracket-spacing

### ISSUE THREAD

**Issue by lo1tuma** [eslint/eslint#3392]
### Problem description

Given the following config

```json
{
    "env": {
        "es6": true
    },
    "ecmaFeatures": {
        "modules": true
    },
    "rules": {
        "comma-spacing": [ 2, { "before": false, "after": true } ],
        "array-bracket-spacing": [ 2, "always" ]
    }
}
```

and the following code:

```js
const [ , a, ...rest ] = foo;
```

then the `comma-spacing` rule warns about an extra space before to first comma in the destructuring statement.

```
foo.js
  1:9  error  There should be no space before ','  comma-spacing

✖ 1 problem (1 error, 0 warnings)
```

when I remove the leading whitespace

```js
const [, a, ...rest ] = foo;
```

then the `array-bracket-spacing` rule warns about a missing space after the opening bracket

```
foo.js
  1:7  error  A space is required after '['  array-bracket-spacing

✖ 1 problem (1 error, 0 warnings)
```

Same problem exist for array literals:

```js
const foo = [ , 1, 2 ];
```

```
foo.js
  1:15  error  There should be no space before ','  comma-spacing

✖ 1 problem (1 error, 0 warnings)
```

I’m using ESLint version 1.1.0

-----

### Possible solution

I suggest that the `comma-spacing` rule should ignore `null` elements of an `ArrayExpression` or `ArrayPattern`.

**Comment by nzakas** [comment]
Not sure you really want to ignore null elements - you'd still want the extra space after, correct?

**Comment by lo1tuma** [comment]
@nzakas You are right, maybe we can just ignore the check for leading extra space of `null` elements?

**Comment by gyandeeps** [comment]
@lo1tuma I actually ended up filtering out the `null` elements in `indent` rule. Maybe we can have a function in `ast-utils` which can do that?

**Comment by lo1tuma** [comment]
@gyandeeps Not sure if we need to filter `null` elements completely in this case. I think we only want to make an exception for the check of leading space.
